### ISSUE TITLE
make error

### ISSUE THREAD

**Issue by jing-wan** [dixonlab/hic_breakfinder#10]
Hi, I try to install hic_beakfinder, 
`
./configure CPPFLAGS="-I /home/wanjing/miniconda2/include/bamtools/ -I /home/wanjing/miniconda2/include/eigen3/" LDFLAGS="-L/home/wanjing/miniconda2/bin/" 
'
But I got error when I make (some part of errors):
'
make

/home/wanjing/miniconda2/include/eigen3/Eigen/src/Core/DenseCoeffsBase.h:56:30: warning: ignoring attributes on template argument ‘Eigen::internal::packet_traits<double>::type {aka __vector(2) double}’ [-Wignored-attributes]
mv -f .deps/hic_breakfinder.Tpo .deps/hic_breakfinder.Po
g++  -g -O2 -g -L/home/wanjing/miniconda2/bin/ -o hic_breakfinder hic_breakfinder.o  
hic_breakfinder.o: In function `get_super_matrix(char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, int, int, float, float)':
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:67: undefined reference to `BamTools::BamReader::BamReader()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:69: undefined reference to `BamTools::BamReader::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:67: undefined reference to `BamTools::BamReader::~BamReader()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:74: undefined reference to `BamTools::BamReader::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:76: undefined reference to `BamTools::BamReader::GetReferenceData() const'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:83: undefined reference to `BamTools::BamAlignment::BamAlignment()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:87: undefined reference to `BamTools::BamReader::GetNextAlignmentCore(BamTools::BamAlignment&)'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:180: undefined reference to `BamTools::BamReader::Close()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:83: undefined reference to `BamTools::BamAlignment::~BamAlignment()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:67: undefined reference to `BamTools::BamReader::~BamReader()'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:83: undefined reference to `BamTools::BamAlignment::~BamAlignment()'
collect2: error: ld returned 1 exit status
make[2]: *** [hic_breakfinder] Error 1
make[2]: Leaving directory `/data3/wanjing/tools/hic_breakfinder/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/data3/wanjing/tools/hic_breakfinder'
make: *** [all] Error 2
'

This is my enviroment:
g++ (GCC) 6.4.0
bamtools 2.4.0
eigen3 3.2.1

Could it caused by the version of g++, bamtools or eigen? Or anny other reasons? Thank you!

**Comment by dixonlab** [comment]
Hello,
I'm not exactly sure where this error comes from, though it doesn't look like it is from eigen. In terms of the paths in your configure command, it seems to say "-L/home/wanjing/miniconda2/bin/". This may not be specific enough to direct the bamtools libraries. Is there a "lib" directory in your "/home/wanjing/miniconda2/include/bamtools" directory?
-Jesse

**Comment by jing-wan** [comment]
Yeah, maybe that's the key. I didn't find the "lib" directory in my bamtools directory. There just are "api" and "shared" directory in my "/home/wanjing/miniconda2/include/bamtools" directory. I tried conda and make to install bamtools and didn't find "lib" directory.

**Comment by jing-wan** [comment]
Hi Jesse, 
I re-installed bamtools and tried again. But I got the similar errors:
`./configure CPPFLAGS="-I /home/wanjing/data3/tools/wshen_build/bamtools-2.4.0/include -I /home/wanjing/miniconda2/include/eigen3/" LDFLAGS="-L/home/wanjing/data3/tools/wshen_build/bamtools-2.4.0/lib"`

make

make  all-recursive
make[1]: Entering directory `/data3/wanjing/tools/hic_breakfinder'
Making all in src
make[2]: Entering directory `/data3/wanjing/tools/hic_breakfinder/src'
g++  -g -O2 -g -L/home/wanjing/data3/tools/wshen_build/bamtools-2.4.0/lib -o hic_breakfinder hic_breakfinder.o  -lbamtools 
hic_breakfinder.o: In function `get_super_matrix(char*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, int, int, float, float)':
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:69: undefined reference to `BamTools::BamReader::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/home/wanjing/data3/tools/hic_breakfinder/src/hic_breakfinder.cpp:74: undefined reference to `BamTools::BamReader::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [hic_breakfinder] Error 1
make[2]: Leaving directory `/data3/wanjing/tools/hic_breakfinder/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/data3/wanjing/tools/hic_breakfinder'
make: *** [all] Error 2

**Comment by jing-wan** [comment]
Hi, I found the error caused by g++ version and lead to bamtools error. I changed g++ from 6.4.0 to  
 4.9.4, and reinstall eigen3 and bamtools. And then it make properly. Hope it can help with other users.

**Comment by jasonsydes** [comment]
Hi, 

I came across this while helping a student trying to get hic_breakfinder installed. I think the trick is to use the versions of bamtools and eigen that were in existence during hic_breakfinder's development and release, which was a bit over three years ago at the time of writing. (I haven't tested extensively, but we also had a bamtools/2.4.0 module that I tried to use, but if memory serves correct, it resulted in the same error you saw).  In any case, for me at least, it looks like bamtools=2.3.0 and eigen=3.3.9 worked.  The following worked for me:

```bash
mamba create -n 20211130_hic_breakfinder bamtools=2.3.0 eigen=3.3.9
conda activate 20211130_hic_breakfinder

make clean
./configure CPPFLAGS="-I /home/sydes/miniconda3/envs/20211130_hic_breakfinder/include/bamtools -I /home/sydes/miniconda3/envs/20211130_hic_breakfinder/include/eigen3/" LDFLAGS="-L/home/sydes/miniconda3/envs/20211130_hic_breakfinder/lib/"
make
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/sydes/miniconda3/envs/20211130_hic_breakfinder/lib

# Testing it, and it works (well, shows the help prompt at least):
∴ ./src/hic_breakfinder
./hic_breakfinder

Required options:

--bam-file [input bam file]
--exp-file-inter [Inter-chromosomal 1Mb expectation file]
--exp-file-intra [Intra-chromosomal 100kb expectation file]
--name [output file name prefix, will append with *.super_matrix.txt and *.SR.txt]

∴
```
Thanks,
Jason
